opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -316,8 +316,8 @@ namespace detail {
         const bool converged = asImpl().getConvergence(timer, iteration);
         const bool must_solve = (iteration < nonlinear_solver.minIter()) || (!converged);
         if (must_solve) {
-            // enable single precision for solvers when dt is smaller then 20 days
-            residual_.singlePrecision = (unit::convert::to(dt, unit::day) < 20.) ;
+            // enable single precision for solvers when dt is smaller then maxSinglePrecisionDays_ (default is 20 days)
+            residual_.singlePrecision = (unit::convert::to(dt, unit::day) < param_.maxSinglePrecisionDays_) ;
```

Comment:
We should keep all unit SI inside the simulator, so I think it is better to make the parameter max_singleprecision_timestep (i.e. not in days), converting from days on input (I agree it is better to ask the user for days than seconds) instead of here.